MPII_Version_* visibility is wrong

I'm trying to compile MPICH at tag `v3.3a2` and I get the following:

```
  CCLD     src/env/mpichversion
  CCLD     src/env/mpivars
src/env/mpichversion.o: In function `main':
/home/daibane/src/mpich/src/env/mpichversion.c:115: undefined reference to `MPII_Version_custom'
/home/daibane/src/mpich/src/env/mpichversion.c:97: undefined reference to `MPII_Version_string'
/home/daibane/src/mpich/src/env/mpichversion.c:109: undefined reference to `MPII_Version_CC'
/home/daibane/src/mpich/src/env/mpichversion.c:110: undefined reference to `MPII_Version_CXX'
/home/daibane/src/mpich/src/env/mpichversion.c:111: undefined reference to `MPII_Version_F77'
/home/daibane/src/mpich/src/env/mpichversion.c:112: undefined reference to `MPII_Version_FC'
/home/daibane/src/mpich/src/env/mpichversion.c:106: undefined reference to `MPII_Version_configure'
/home/daibane/src/mpich/src/env/mpichversion.c:103: undefined reference to `MPII_Version_device'
/home/daibane/src/mpich/src/env/mpichversion.c:100: undefined reference to `MPII_Version_date'
collect2: error: ld returned 1 exit status
```

any ideas how I could debug this ?

--- a/src/include/mpir_misc.h
+++ b/src/include/mpir_misc.h
@@ -48,15 +48,15 @@ typedef enum MPIR_Lang_t {
 #endif
 } MPIR_Lang_t;
 
-extern const char MPII_Version_string[];
-extern const char MPII_Version_date[];
-extern const char MPII_Version_configure[];
-extern const char MPII_Version_device[];
-extern const char MPII_Version_CC[];
-extern const char MPII_Version_CXX[];
-extern const char MPII_Version_F77[];
-extern const char MPII_Version_FC[];
-extern const char MPII_Version_custom[];
+extern const char MPII_Version_string[] MPICH_API_PUBLIC;
+extern const char MPII_Version_date[] MPICH_API_PUBLIC;
+extern const char MPII_Version_configure[] MPICH_API_PUBLIC;
+extern const char MPII_Version_device[] MPICH_API_PUBLIC;
+extern const char MPII_Version_CC[] MPICH_API_PUBLIC;
+extern const char MPII_Version_CXX[] MPICH_API_PUBLIC;
+extern const char MPII_Version_F77[] MPICH_API_PUBLIC;
+extern const char MPII_Version_FC[] MPICH_API_PUBLIC;
+extern const char MPII_Version_custom[] MPICH_API_PUBLIC;
 
 int MPIR_Localcopy(const void *sendbuf, MPI_Aint sendcount, MPI_Datatype sendtype,
                    void *recvbuf, MPI_Aint recvcount, MPI_Datatype recvtype);

--- a/src/mpi/init/initinfo.c
+++ b/src/mpi/init/initinfo.c
@@ -4,7 +4,7 @@
  *      See COPYRIGHT in top-level directory.
  */
 
-#include "mpi.h"
+#include "mpiimpl.h"
 #include "mpichinfo.h"
 /* 
    Global definitions of variables that hold information about the

